File: eval/fixtures/rails-pms/app/models/reservation.rb
# A single reservation.
#
# Guest contact details are stored inline (no separate accounts / login) to keep
# the PMS minimal. Dates are ISO YYYY-MM-DD strings, which sort as dates.
class Reservation < ApplicationRecord
  STATUSES = %w[pending confirmed cancelled].freeze

  # Nullable: a stay can be taken before the front desk has decided which space
  # the guest gets. Clearing the space on delete keeps the reservation valid.
  belongs_to :space, optional: true

  validates :guest_name, presence: true
  validates :email, presence: true, format: { with: URI::MailTo::EMAIL_REGEXP }
  validates :phone, presence: true, length: { minimum: 5 }
  validates :check_in, presence: true
  validates :check_out, presence: true
  validates :party_size,
            numericality: {
              only_integer: true,
              greater_than_or_equal_to: 1,
              less_than_or_equal_to: 20,
            }
  validates :status, inclusion: { in: STATUSES }
  validate :check_out_after_check_in

  scope :newest_first, -> { order(created_at: :desc) }

  # All reservations, newest first, with the assigned space eager-loaded.
  def self.list
    newest_first.includes(:space)
  end

  private

  def check_out_after_check_in
    return if check_in.blank? || check_out.blank?

    errors.add(:check_out, "must be after check-in") if check_out <= check_in
  end
end

File: eval/fixtures/laravel-pms/app/Models/Reservation.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

/**
 * A single reservation.
 *
 * Guest contact details are stored inline (no separate accounts / login) to
 * keep the PMS minimal. Check-in / check-out are ISO YYYY-MM-DD strings, which
 * sort as dates.
 *
 * @property int $id
 * @property string $guest_name
 * @property string $email
 * @property string $phone
 * @property string $check_in
 * @property string $check_out
 * @property int $party_size
 * @property string|null $notes
 * @property int|null $space_id
 * @property string $status
 */
class Reservation extends Model
{
    /** Only a created_at timestamp — the domain has no updated_at. */
    const UPDATED_AT = null;

    protected $fillable = [
        'guest_name',
        'email',
        'phone',
        'check_in',
        'check_out',
        'party_size',
        'notes',
        'space_id',
        'status',
    ];

    protected $attributes = [
        'party_size' => 1,
        'status' => 'pending',
    ];

    protected function casts(): array
    {
        return [
            'party_size' => 'integer',
            'space_id' => 'integer',
            'created_at' => 'datetime',
        ];
    }

    /**
     * Assigned space. Nullable: a stay can be taken before the front desk has
     * decided which space the guest gets.
     *
     * @return BelongsTo<Space, $this>
     */
    public function space(): BelongsTo
    {
        return $this->belongsTo(Space::class);
    }
}
